include/vcl/Video.h

```diff
@@ -256,10 +261,279 @@ namespace VCL {
      */
     void store(const std::string &video_id, Codec video_codec);
 
+    /**
+     *  Stores a Write Operation in the list of operations, performs the
+     *  operations that are already in the list, and finally writes the
+     *  video to the disk.
+     *  This method will used when the video_id and codec are already defined.
+     */
+    void store();
+
     /**
      *  Deletes the Video file
      */
     void delete_video();
+
+    private:
+
+        class Operation;
+
+        // Full path to the video file.
+        // It is called _video_id to keep it consistent with VCL::Image
+        std::string _video_id;
+
+        bool _flag_stored; // Flag to avoid unnecessary read/write
+
+        std::vector<VCL::Image> _frames;
+
+        VideoSize _size;
+
+        float _fps;
+
+        Codec _codec; // (h.264, etc).
+
+        std::vector<std::shared_ptr<Operation>> _operations;
+
+    /*  *********************** */
+    /*        OPERATION         */
+    /*  *********************** */
+
+        /**
+         *  Provides a way to keep track of what operations should
+         *   be performed on the data when it is needed
+         *
+         *  Operation is the base class, it keeps track of the format
+         *   of the Video data, defines a way to convert Format to
+         *   a string, and defines a virtual function that overloads the
+         *   () operator
+         */
+        class Operation {
```

Comment:
Similar comment as in Image operations.

Reply:
Ditto for Image operations: I suggest we do that in a separate patch.